bin.src/translate_header.py

```diff
@@ -5,48 +5,81 @@
 import re
 import sys
 import traceback
+import importlib
 import yaml
-from astro_metadata_translator import ObservationInfo
+from astro_metadata_translator import ObservationInfo, merge_headers
 
 # Prefer afw over Astropy
 try:
-    from lsst.afw.fits import readMetadata as read_metadata  # noqa: N813
+    from lsst.afw.fits import readMetadata
     import lsst.daf.base  # noqa: F401 need PropertyBase for readMetadata
+
+    def read_metadata(file, hdu=0):
+        try:
+            return readMetadata(file, hdu=hdu)
+        except lsst.afw.fits.FitsError:
+            return None
+
 except ImportError:
     from astropy.io import fits
 
-    def read_metadata(file, hdu=1):
+    def read_metadata(file, hdu=0):
         fits_file = fits.open(file)
-        return fits_file[hdu].header
+        try:
+            header = fits_file[hdu].header
+        except IndexError:
+            header = None
+        return header
 
 parser = argparse.ArgumentParser(description="Summarize headers from astronomical data files")
 parser.add_argument("files", metavar="file", type=str, nargs="+",
                     help="File(s) from which headers will be parsed."
                     " If a directory is given it will be scanned for files matching the regular"
                     " expression defined in --regex.")
-parser.add_argument("-q", "--quiet", const=True, default=False, action="store_const",
+parser.add_argument("-q", "--quiet", action="store_true",
                     help="Do not report the translation content from each header")
-parser.add_argument("-d", "--dumphdr", const=True, default=False, action="store_const",
+parser.add_argument("-d", "--dumphdr", action="store_true",
                     help="Dump the header in YAML format to standard output rather than translating it")
-parser.add_argument("--traceback", const=True, default=False, action="store_const",
+parser.add_argument("--traceback", action="store_true",
                     help="Give detailed trace back when any errors encountered")
+parser.add_argument("-n", "--hdrnum", default=1,
```

Comment:
The one default here is that most files don't have a full header in the 0th one and it needs merging. I don't ever call `read_metadata` with the default parameter so it may be simpler to remove the defaulting completely.